web/src/containers/P2P/TradesHistory/index.tsx

```diff
@@ -48,29 +50,56 @@ const TradesHistory: FC = (): ReactElement => {
         dispatch(p2pTradesHistoryFetch({ page: Number(page) + 1, limit: DEFAULT_TABLE_PAGE_LIMIT }));
     }, [p2pTradesHistoryFetch, page, DEFAULT_TABLE_PAGE_LIMIT]);
 
-    const renderHeaders = () => {
+    const renderHeaders = useMemo(() => {
         return [
-            intl.formatMessage({id: 'page.body.p2p.order_history.date'}),
-            intl.formatMessage({id: 'page.body.p2p.order_history.side'}),
-            intl.formatMessage({id: 'page.body.p2p.order_history.price'}),
-            intl.formatMessage({id: 'page.body.p2p.order_history.quantity'}),
-            intl.formatMessage({id: 'page.body.p2p.order_history.counterparty'}),
-            intl.formatMessage({id: 'page.body.p2p.order_history.status'}),
+            translate('page.body.p2p.order_history.date'),
+            translate('page.body.p2p.order_history.side'),
+            translate('page.body.p2p.order_history.price'),
+            translate('page.body.p2p.order_history.quantity'),
+            translate('page.body.p2p.order_history.counterparty'),
+            translate('page.body.p2p.order_history.status'),
         ];
-    };
+    }, [translate]);
+
+    const renderTableRow = useCallback((item: P2POrder) => {
+        const { created_at, side, amount, state, user_uid } = item;
+        const { price, quote, base, user, uid } = item.offer;
+        const sideColored = setTradesType(side);
+        const stateColored = setStateType(state);
+        const pricePrecision = currencies.find(item => item.id === quote)?.precision;
+        const amountPrecision = currencies.find(item => item.id === base)?.precision;
 
-    const renderContent = () => {
         return [
+            localeDate(created_at, 'fullDate'),
+            <span style={{ color: sideColored.color }}>{sideColored.text}</span>,
+            `${Decimal.format(price, pricePrecision || DEFAULT_FIAT_PRECISION, ',')} ${base.toUpperCase()}/${quote.toUpperCase()}`,
+            `${Decimal.format(amount, amountPrecision || DEFAULT_CCY_PRECISION, ',')} ${base.toUpperCase()}`,
+            <span>{user?.user_nickname || uid}</span>,
+            <span style={{ color: stateColored.color }}>{stateColored.text}</span>,
+        ];
+    }, [currencies, DEFAULT_FIAT_PRECISION, DEFAULT_CCY_PRECISION, setTradesType, setStateType]);
+
+    const retrieveData = useMemo(() => {
+        return [...list].map(item => renderTableRow(item));
```

Comment:
Why do you use ...list?
Do list.map(...)

Reply:
Fixed, thanks.